gcloud-java-examples/src/main/java/com/google/gcloud/examples/DnsExample.java

```diff
@@ -0,0 +1,516 @@
+      Iterator<ChangeRequest> iterator;
+      if (args.length > 2) {
+        Dns.SortingOrder sortOrder = Dns.SortingOrder.valueOf(args[2].toUpperCase());
+        iterator = dns.listChangeRequests(zoneName,
+            Dns.ChangeRequestListOption.sortOrder(sortOrder)).iterateAll();
+      } else {
+        iterator = dns.listChangeRequests(zoneName).iterateAll();
+      }
+      if (iterator.hasNext()) {
+        System.out.printf("Change requests for zone %s:%n", zoneName);
+        System.out.printf("ID\tStatus\tTimestamp%n");
+        while (iterator.hasNext()) {
+          ChangeRequest change = iterator.next();
+          System.out.printf("%s\t%s\t%s%n", change.id(), change.status(), change.startTimeMillis());
+          System.out.printf("\tDeletions: %s%n", Joiner.on(",").join(change.deletions()));
+          System.out.printf("\tAdditions: %s%n", Joiner.on(",").join(change.additions()));
+        }
+      } else {
+        System.out.printf("Zone %s has no change requests.%n", zoneName);
+      }
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name> changes [descending | ascending]";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      System.err.println(Arrays.asList(args));
+      return args.length == 2
+          || (args.length == 3 && ImmutableList.of("descending", "ascending").contains(args[2]));
+    }
+  }
+
+  private static class ListAction implements DnsAction {
+
+    /**
+     * Invokes a list action. If no parameter is provided, lists all zones. If zone name is the only
+     * parameter provided, lists both DNS records and changes. Otherwise, invokes listing changes or
+     * zones based on the parameter provided.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      if (args.length == 0) {
+        new ListZonesAction().run(dns);
+      } else {
+        if (args.length == 1 || "records".equals(args[1])) {
+          new ListDnsRecordsAction().run(dns, args);
+        }
+        if (args.length == 1 || "changes".equals(args[1])) {
+          new ListChangesAction().run(dns, args);
+        }
+      }
+    }
+
+    @Override
+    public boolean check(String... args) {
+      if (args.length == 0 || args.length == 1) {
+        return true;
+      }
+      if ("records".equals(args[1])) {
+        return new ListDnsRecordsAction().check(args);
+      }
+      if ("changes".equals(args[1])) {
+        return new ListChangesAction().check(args);
+      }
+      return false;
+    }
+
+    @Override
+    public String params() {
+      return "[<zone_name> [changes [descending | ascending] | records]]";
+    }
+  }
+
+  private static class GetProjectAction implements DnsAction {
+
+    @Override
+    public void run(Dns dns, String... args) {
+      ProjectInfo project = dns.getProject();
+      System.out.printf("Project id: %s%nQuota:%n", dns.options().projectId());
+      System.out.printf("\tZones: %d%n", project.quota().zones());
```

Comment:
Should probably save `project.quota()` in a variable to avoid repeating it below.